hack/local-up-kubeedge.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+KUBEEDGE_ROOT=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/..
+ENABLE_DAEMON=${ENABLE_DAEMON:-false}
+LOG_DIR=${LOG_DIR:-"/tmp"}
+
+
+if [[ "${CLUSTER_NAME}x" == "x" ]];then
+    CLUSTER_NAME="test"
+fi
+
+export CLUSTER_CONTEXT="--name ${CLUSTER_NAME}"
+
+# spin up cluster with kind command
+function kind-up-cluster {
+  check-prerequisites
+  check-kind
+  echo "Running kind: [kind create cluster ${CLUSTER_CONTEXT} ${KIND_OPT}]"
+  kind create cluster ${CLUSTER_CONTEXT}
+}
+
+function uninstall-kubeedge {
+  # kill the cloudcore
+  [[ -n "${CLOUDCORE_PID-}" ]] && sudo kill "${CLOUDCORE_PID}" 2>/dev/null
+
+  # kill the edgecore
+  [[ -n "${EDGECORE_PID-}" ]] && sudo kill "${EDGECORE_PID}" 2>/dev/null
+}
+
+# clean up
+function cleanup {
+  echo "Cleaning up..."
+  uninstall-kubeedge
+
+  echo "Running kind: [kind delete cluster ${CLUSTER_CONTEXT}]"
+  kind delete cluster ${CLUSTER_CONTEXT}
+}
+
+if [[ "${ENABLE_DAEMON}" = false ]]; then
+  trap cleanup EXIT
+fi
+
+function create-device-crd {
+  echo "creating the device crd..."
+  kubectl apply -f ${KUBEEDGE_ROOT}/build/crds/devices/devices_v1alpha1_device.yaml
+  kubectl apply -f ${KUBEEDGE_ROOT}/build/crds/devices/devices_v1alpha1_devicemodel.yaml
+}
+
+function build-cloudcore {
+  # TODO: make the binary to _output dir
+  echo "building the cloudcore..."
+  make -C "${KUBEEDGE_ROOT}" WHAT="cloud"
+  mkdir -p ${KUBEEDGE_ROOT}/_output/bin/cloud
+  mv ${KUBEEDGE_ROOT}/cloud/cloudcore ${KUBEEDGE_ROOT}/_output/bin/cloud
+  cp -r ${KUBEEDGE_ROOT}/cloud/conf ${KUBEEDGE_ROOT}/_output/bin/cloud
```

Comment:
TODO: let the make generate example yaml file under target output folder.

Reply:
sure, will do later.